### PR TITLE
initial support for upstream 36

### PR DESCRIPTION
**Pull request by richm**
Rename README-mux.md to README-install.md since mux is now part of openshift.  Updated instructions for latest 3.6 candidate origin upstream.
@sradco ptal

### CODE REVIEW

File: README-install.md [hunk 1]
```diff
@@ -152,6 +147,10 @@ determine using ansible, you may need to change the following fields in
   passwordless ssh
 * `ansible_become` - use `no` if `ansible_ssh_user` is `root`, otherwise,
   use `yes`
+* `openshift_logging_mux_namespaces` - **REQUIRED** Represents the environment
```
Comment (nhosoi) [inline README-install.md:153]: According to the line 191, REQUIRED may be a little too strong?
It'd be easier to understand if the line 191-192 is moved next to the line 153?

File: README-install.md [hunk 2]
```diff
@@ -266,22 +262,10 @@ This will create the `logging-mux` configmap, secrets, dc, pod, and service.
...
-
-    # MUX_HOST=mux.logging.test MUX_PUBLIC_IP=192.168.122.4 \
-      MUX_NAMESPACES="my-first-namespace my-other-ns" ./setup-mux.sh
-
-You can test with `openssl s_client` like this:
+whichever interface on the machine is used for external egress, that you
+configured with the `openshift_ip` in Ansible.  You can test with `openssl
+s_client` like this:
 
     # echo hello | openssl s_client -quiet -connect localhost:24284
```
Comment (nhosoi) [inline README-install.md:278]: Trailing \ is needed?

File: vars.yaml.template [hunk 3]
```diff
@@ -26,3 +29,16 @@ openshift_logging_master_public_url: https://{{ openshift_public_hostname }}:844
...
+openshift_logging_es_hostname: es.{{ openshift_master_default_subdomain }}
+
+# the public hostname for common logging ingestion - the fluentd secure_forward listener
+openshift_logging_mux_hostname: mux.{{ openshift_master_default_subdomain }}
+
+# mux tuning parameters
+#openshift_logging_mux_cpu_limit: 500m
+#openshift_logging_mux_memory_limit: 2Gi
+#openshift_logging_mux_buffer_queue_limit: 1024
+#openshift_logging_mux_buffer_size_limit: 1m
```
Comment (nhosoi) [inline vars.yaml.template]: Maybe, 16m is preferable?

**Comment by nhosoi** [comment]
@sradco, @richm, I'm pasting the Shirly's questions to this conversation page.
On 08/06/2017 04:04 AM, Shirly Radco wrote:
> https://github.com/richm/Main/blob/649e019b7c962b24ea96478fe69bfce02b83dc90/README-install.md#provisioning-a-machine-to-run-viaq
>
>     assign the machine an FQDN and IP address so that it can be reached from another machine - these are the public_hostname and public_ip
>     use root (or create a user account) - this user will be referred to below as $USER
>     provide an ssh pubkey for this user account (ssh-keygen)
>     add the ssh pubkey to the user account $USER/.ssh/authorized_keys
>         cat $USER/.ssh/id_rsa.pub >> $USER/.ssh/authorized_keys
>     add the ssh hostkey for localhost to your SSH known_hosts
>         ssh-keyscan -H localhost >> $USER/.ssh/known_hosts
>     add the ssh hostkey for public_hostname to your SSH known_hosts
>         ssh-keyscan -H **public_hostname** >> $USER/.ssh/known_hosts
>     if not using root, enable passwordless sudo e.g. in sudoers config:
>         $USER ALL=(ALL) NOPASSWD:ALL
>     allow connections on the following ports/protocols:
>         icmp (for ping)
>         tcp ports 22, 80, 443, 8443 (openshift console), 24284 (secure_forward)
>
> In this section it is not clear on which machine the user should do the steps described.

This section is the beginning.  So, it may be helpful to describe the environment we are trying to deploy the logging system on.   And what we will get once the listed steps are successfully done.  I think there are certainly various use cases.  For instance, in the RHV env, you may assign one VM to install the logging system in which elasticsearch, mux, kibana, and other pods run?   We may want to mention the cloud centric environment, as well.

> Also, If I understand correctly, not all of them are required when using user root. So please write a remark what is relevant only if not using root.

Please correct me if I'm wrong, I think just this sudo step is not necessary for the user root.
if not using root, enable passwordless sudo e.g. in sudoers config:
    $USER ALL=(ALL) NOPASSWD:ALL

> Also $USER is not equal to root or user name. It is equal to /home/user right?

I read $USER is a **user name** of the created user account or root from this sentence...
"use root (or create a user account) - this user will be referred to below as $USER"

**Comment by sradco** [comment]
Hi,

I updated the rep.
I got this error when running ansible

Failure summary:

  1. Host:     localhost
     Play:     Verify Requirements
     Task:     openshift_health_check
     Message:  One or more checks failed
     Details:  check "docker_image_availability":
               Some dependencies are required in order to check Docker image availability.
               Repository centos-openshift-origin is listed more than once in the configuration
               warning: /var/cache/yum/x86_64/7/centos-extras/packages/python-docker-pycreds-1.10.6-1.el7.noarch.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
               
               
               Public key for python-docker-pycreds-1.10.6-1.el7.noarch.rpm is not installed


I needed to update [centos-extras] gpgcheck to 0.

**Comment by sradco** [comment]
Run failed with this message: 

MSG:

Unable to start service docker: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.


	to retry, use: --limit @/usr/share/ansible/openshift-ansible/playbooks/byo/config.retry

PLAY RECAP ***********************************************************************************************************************************************************************************
localhost                  : ok=222  changed=9    unreachable=0    failed=1   


Failure summary:

  1. Host:     localhost
     Play:     Configure masters
     Task:     docker : Start the Docker service
     Message:  Unable to start service docker: Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.


Updated node-config.yaml with these args 

kubeletArguments:
  cgroup-driver:
    - 'cgroupfs'

**Comment by sradco** [comment]
After a very very long run the ansible failed with this error:

TASK [openshift_logging_mux : Add mux namespaces] ********************************************************************************************************************************************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_logging_mux/tasks/main.yaml:202
fatal: [localhost]: FAILED! => {
    "failed": true
}

MSG:

Unexpected templating type error occurred on ({{ openshift_logging_mux_namespaces | union(openshift_logging_mux_default_namespaces) }}): coercing to Unicode: need string or buffer, list found

	to retry, use: --limit @/usr/share/ansible/openshift-ansible/playbooks/byo/config.retry

PLAY RECAP ***********************************************************************************************************************************************************************************
localhost                  : ok=713  changed=109  unreachable=0    failed=1   


Failure summary:

  1. Host:     localhost
     Play:     Create Hosted Resources
     Task:     openshift_logging_mux : Add mux namespaces
     Message:  Unexpected templating type error occurred on ({{ openshift_logging_mux_namespaces | union(openshift_logging_mux_default_namespaces) }}): coercing to Unicode: need string or buffer, list found




My vars.yaml file include :

openshift_logging_mux_namespaces: "project.ovirt-metrics-engine project.ovirt-logs-engine"


How should my openshift_logging_mux_namespaces be?
I keep failing on this issue.
@richm 

Perhaps we should add an example of a namespace or add guidelines to it and how to add more than one.

**Comment by richm** [comment]
@sradco How did you install this centos system?  Do you have the centos-release package?  This package is what provides the centos base and extras repo (so we shouldn't have to include them in the viaq repo list) - `rpm -q centos-release` and `rpm -ql centos-release`

You should not have to update node-config.yaml - I'm not sure what docker is complaining about.

**Comment by sradco** [comment]
I installed it from PXE.

[root@sradco-metrics-1 openshift-ansible]# rpm -q centos-release
centos-release-7-3.1611.el7.centos.x86_64
[root@sradco-metrics-1 openshift-ansible]# rpm -ql centos-release
/etc/centos-release
/etc/centos-release-upstream
/etc/issue
/etc/issue.net
/etc/os-release
/etc/pki/rpm-gpg
/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Debug-7
/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Testing-7
/etc/redhat-release
/etc/rpm/macros.dist
/etc/system-release
/etc/system-release-cpe
/etc/yum.repos.d/CentOS-Base.repo
/etc/yum.repos.d/CentOS-CR.repo
/etc/yum.repos.d/CentOS-Debuginfo.repo
/etc/yum.repos.d/CentOS-Media.repo
/etc/yum.repos.d/CentOS-Sources.repo
/etc/yum.repos.d/CentOS-Vault.repo
/etc/yum.repos.d/CentOS-fasttrack.repo
/etc/yum/vars/infra
/usr/lib/systemd/system-preset/85-display-manager.preset
/usr/lib/systemd/system-preset/90-default.preset
/usr/share/centos-release/EULA
/usr/share/doc/centos-release/Contributors
/usr/share/doc/centos-release/GPL
/usr/share/doc/redhat-rele

**Comment by richm** [comment]
can you paste the contents of `/etc/yum.repos.d/CentOS-Base.repo` ?

**Comment by sradco** [comment]
I get "No resources found." for "oc get pods -l component=mux"


[root@sradco-metrics-1 openshift-ansible]# oc get pods -l component=mux
No resources found.
[root@sradco-metrics-1 openshift-ansible]# oc get pods
NAME                                      READY     STATUS    RESTARTS   AGE
logging-curator-1-jz4c5                   1/1       Running   2          5h
logging-es-data-master-v0436hru-1-cdq3t   1/1       Running   2          5h
logging-fluentd-z3srk                     1/1       Running   0          6m
logging-kibana-1-t8mfz                    2/2       Running   4          5h
logging-mux-1-deploy                      0/1       Error     0          5h
@richm @nhosoi

**Comment by sradco** [comment]
Sorry, I only had 
[CentOS-7.3-updates]
name=CentOS-7.3-updates
baseurl=http://download.eng.tlv.redhat.com/pub/centos/released/7.3.1611/updates/x86_64/
enabled=1
priority=99
gpgcheck=0


[CentOS-7.3]
name=CentOS-7.3
baseurl=http://download.eng.tlv.redhat.com/pub/centos/released/7.3.1611/os/x86_64/
enabled=1
priority=99
gpgcheck=0

Thank you.

**Comment by sradco** [comment]
When running:
[root@sradco-metrics-1 openshift-ansible]# oc get secret logging-mux --template='{{index .data "mux-ca"}}' | base64 -d > mux-ca.crt
base64: invalid input 

[root@sradco-metrics-1 openshift-ansible]# oc get secret logging-mux --template='{{index .data "mux-ca"}}'
<no value>

It seems that the ansible script did not create the mux-ca.crt

@richm @nhosoi

**Comment by richm** [comment]
@sradco @nhosoi I'm not sure what to do about the centos repo problem.  I think viaq should not include any centos base or extras repos, and should instead assume that centos is setup as per the centos setup instructions.  If those centos instructions do not specify that extras must be enabled by default, then I will add them to the viaq instructions.

**Comment by sradco** [comment]
I think it is not necessary to add it to the viaq documentation. @richm @nhosoi

**Comment by richm** [comment]
Looks like mux did not deploy properly - no pod, no cert.  I'm investigating now.

**Comment by richm** [comment]
@nhosoi @sradco please check that I have addressed all of the review comments/questions with my latest commits

**Comment by nhosoi** [comment]
Looks good to me.

A minor thing.  There are 2 files which refer setup-mux.sh.  Does this mean run-from-beaker.sh needs to be updated to use ansible?
```
README-json.md:could use the old setup-mux.sh script to hack something together.
run-from-beaker.sh:if [ -x $HOME/ViaQ/setup-mux.sh ] ; then
run-from-beaker.sh:    MUX_HOST=mux.$hostname $HOME/ViaQ/setup-mux.sh
```

**Comment by richm** [comment]
@nhosoi no - the expectation is that if `$HOME/ViaQ/setup-mux.sh` exists, it will be used.  This PR removes setup-mux.sh from the branch.  That way, we don't have to change run-from-beaker.sh for each branch.

**Comment by nhosoi** [comment]
Thanks for the clarification about setup-mux.sh, @richm!

**Comment by tasdikrahman** [comment]
Any updates on this PR?

**Pull request merged**